### DISCUSSION TITLE
Get a list of task names

### DISCUSSION THREAD

**Discussion by HendrikSchrieber** [argoproj/argo-workflows#12715] [Q&A]
Hey everybody,

is it possible to somehow get a list of the names of all tasks of a workflow from within that workflow? I know that I can get certain information using [workflow variables](https://argo-workflows.readthedocs.io/en/stable/variables/#dag-templates) but those only get me the name of a single task - the one that is currently being executed. I need a list of all tasks names so I can pass it to an external CI system (e.g. GitLab) and sync the pipeline status.

Thanks!
Hendrik

**Comment by wesleyscholl** [comment]
@HendrikSchrieber 

Currently, Argo Workflows does not provide a built-in way to list all task names in a workflow dynamically from within the workflow itself. Workflow variables such as `{{tasks.<TASKNAME>.status}}` or `{{tasks.<TASKNAME>.id}}` give access to specific task information, but they cannot retrieve all task names at once.

https://argo-workflows.readthedocs.io/en/latest/variables/#dag-templates

However, you could extract DAG task names using `kubectl` and `jq`.

```bash
kubectl get workflow <workflow-name> -o json | jq -r '.spec.templates[] | select(.dag.tasks) | .dag.tasks[].name'
```
Example Output

If your Workflow contains tasks like task-1, task-2, and task-3, the above commands will return:

```bash
task-1
task-2
task-3
```

References
[Argo Workflows Documentation](https://argo-workflows.readthedocs.io/en/latest/)
[Kubernetes CLI Reference](https://kubernetes.io/docs/reference/kubectl/quick-reference/)
